### PR TITLE
#184581332 Login with google account 

### PR DESCRIPTION
**Pull request by Habihirwe**
**What does this PR do?**

allows user to login with google account

**Description of Task to be completed?**

first we check wether the email of the user is in data base 
if yes  it displays the success message, the token , and the role of user.

**endpoints:**

/auth/auth/google/callback

**How should this be manually tested?**

after cloning the repo,
you have to create the client credentials
so that you can get 
-clientID
-clientSecret
-callbackURL
in terminal add npm install

- add .env in your repo and assign the variables as there are in .env.example file
- run in termainal npm run migrate:all to setup the database
- run in termainal npm start to start the project
- in your browser add http://localhost:(your port)/auth/google

**Any background context you want to provide?**
n/a

**What are the relevant pivotal tracker/Trello stories?**

https://www.pivotaltracker.com/story/show/184581332

**Screenshots (if appropriate)**

**Questions:**
n/a

### CODE REVIEW

File: src/routes/auth.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    
+);
+
+route.get('/profile', successGoogleLogin);
+
+async function successGoogleLogin(req, res) {
+  if (!req.user) return res.redirect("/auth/login");
+  try {
+      const userObj = req.user._json;
+      if (userObj.email_verified != true) return res.status(401).json({ message: "Email not verified", error: true });
```
Comment (blackd44) [inline src/routes/auth.js]: can you change `userObj.email_verified` to `userObj.verified` in order to much with others tasks as:

Screen Shot 2023-03-14 at 14 53 59

File: src/routes/auth.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          delete newUser.password;
+          const token = jwt.sign(newUser, process.env.SECRET_KEY);
+          res.status(200).json({
+              message: 'success',
+              token,
+              role: user.role
+          });
+      }
+  } catch (err) {
+      console.error(err);
```
Comment (blackd44) [inline src/routes/auth.js]: remove `console`

**Review by blackd44 — CHANGES_REQUESTED** [review]
Others Looks Good

File: src/routes/auth.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+dotenv.config();
+const route = express.Router();
+passport.serializeUser((user, done) => { done(null, user) })
+
+passport.deserializeUser((user, done) => { done(null, user) })
+
+passport.use(
+    new GoogleStrategy(
+        {
+            clientID: "1072588531381-nmcmlfa0m0a2tmu5lqh18bc1lrec23t9.apps.googleusercontent.com",
```
Comment (nshutijonathan) [inline src/routes/auth.js]: this is sensitive information and should be loaded from the `env` file

File: src/routes/auth.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            clientSecret:"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" ,
```
Comment (nshutijonathan) [inline src/routes/auth.js]: clientSecret should not be exposed to the public

**Review by nshutijonathan — CHANGES_REQUESTED** [review]
The title of your PR is not the right one ,please correct is

**Review by Patienceineza — APPROVED** [review]
LGTM

**Review by edwinnambaje — APPROVED** [review]
[REDACTED]

**Review by edwinnambaje — CHANGES_REQUESTED** [review]
fix changes you had been requested to do and also conflicts with the tests

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by Zeden12 — APPROVED** [review]
LGTM

**Comment by Zeden12** [comment]
fix the conflicts with the tests and you're ready to move

**Review by Martha-Iradukunda — COMMENTED** [review]
LGTM

**Review by Martha-Iradukunda — APPROVED** [review]

**Review by edwinnambaje — APPROVED** [review]
LGTM

**Review by blackd44 — APPROVED** [review]

**Comment by blackd44** [comment]
solve those conflicts

**Review by INValentin — APPROVED** [review]
LGTM

**Review by UmuhireJessie — APPROVED** [review]

**Pull request merged**